packages/x-data-grid-premium/src/hooks/features/pivoting/gridPivotingInterfaces.ts

```diff
@@ -0,0 +1,85 @@
+import type { GridColDef } from '@mui/x-data-grid-pro';
+import type { GridPivotingStatePartial } from '@mui/x-data-grid/internals';
+import type { DataGridPremiumProcessedProps } from '../../../models/dataGridPremiumProps';
+
+export type GridPivotingPropsOverrides = {
+  rows: DataGridPremiumProcessedProps['rows'];
+  columns: DataGridPremiumProcessedProps['columns'];
+  rowGroupingModel: NonNullable<DataGridPremiumProcessedProps['rowGroupingModel']>;
+  aggregationModel: NonNullable<DataGridPremiumProcessedProps['aggregationModel']>;
+  getAggregationPosition: NonNullable<DataGridPremiumProcessedProps['getAggregationPosition']>;
+  columnVisibilityModel: NonNullable<DataGridPremiumProcessedProps['columnVisibilityModel']>;
+  columnGroupingModel: NonNullable<DataGridPremiumProcessedProps['columnGroupingModel']>;
+  groupingColDef: NonNullable<DataGridPremiumProcessedProps['groupingColDef']>;
+};
```

Comment:
It would be helpful to add a callout on the docs mentioning that these props if provided by the user will not be applied in the pivot mode as the pivoting logic computes them and overrides over the user-provided ones.

Reply:
Side note: Implementing [#10861](https://github.com/mui/mui-x/issues/10861) will require significant changes to the server-side logic, as column computation will be shifted to the server, unlike the current approach.

Reply:
> Implementing https://github.com/mui/mui-x/issues/10861 will require some changes to the server-side logic, as column computation will be shifted to the server

Yes, columns must be part of the server response every time the pivot model changes.